src/MetricsUploader/include/MetricsUploader/MetricsUploader.h

```diff
@@ -36,39 +36,20 @@ const std::string kMetricsUploaderClientDllName = "metrics_uploader_client";
 //
 class MetricsUploader {
  public:
+  MetricsUploader() = default;
+  virtual ~MetricsUploader() = default;
+
   // Create a MetricsUploader instance, load metrics uploader client library if available and starts
   // metrics uploader client when called first time. Return the instance if there are no errors and
   // ErrorMessage otherwise.
-  [[nodiscard]] static ErrorMessageOr<MetricsUploader> CreateMetricsUploader(
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<MetricsUploader>> CreateMetricsUploader(
       std::string client_name = kMetricsUploaderClientDllName);
 
-  // Unload metrics_uploader_client.dll
-  ~MetricsUploader();
-
-  MetricsUploader(const MetricsUploader& other) = delete;
-  MetricsUploader(MetricsUploader&& other);
-  MetricsUploader& operator=(const MetricsUploader& other) = delete;
-  MetricsUploader& operator=(MetricsUploader&& other);
-
   // Send log events to the server using metrics_uploader.
   // Returns true on success and false otherwise.
-  bool SendLogEvent(OrbitLogEvent_LogEventType log_event_type,
-                    std::chrono::milliseconds event_duration = std::chrono::milliseconds::zero());
-
- private:
-#ifdef _WIN32
-  HMODULE metrics_uploader_client_dll_;
-  explicit MetricsUploader(std::string client_name, std::string session_uuid,
-                           Result (*send_log_event_addr)(const uint8_t*, int),
-                           Result (*shutdown_connection)(), HMODULE metrics_uploader_client_dll);
-#else
-  MetricsUploader();
-#endif
-
-  Result (*send_log_event_addr_)(const uint8_t*, int) = nullptr;
-  Result (*shutdown_connection_addr_)() = nullptr;
-  std::string client_name_;
-  std::string session_uuid_;
+  virtual bool SendLogEvent(
+      OrbitLogEvent_LogEventType log_event_type,
+      std::chrono::milliseconds event_duration = std::chrono::milliseconds::zero()) = 0;
```

Comment:
Maybe we can add base implementation for this function instead of making it pure virtual? In this case we can return from CreateMetricsUploader the base class instance if we don't have `.dll` available (or in case other errors), and avoid checking `MetricsUploader*` for nullptr when we want to send metrics.

Reply:
I am going to be quite honest, I wasn't sure if this is a good idea or not, so I talked to Henning about it. He confirmed to me that the current implementation with `MetricsUploader` only being an interface is actually pretty nice. For the reason that if someone else wants to write another child class to `MetricsUploader`, there will be compiler errors if it was forgotten to implement a method (in this case only `SendLogEvent`). If it's not a pure virtual class, it could happen that the base implementation is accidentally used, because there are no errors.  

I hope this explanation makes sense.

Conclusion: I would leave everything as is, wdyt?